src/compiler/preprocess/index.ts

```diff
@@ -2,6 +2,7 @@ import { RawSourceMap, DecodedSourceMap } from '@ampproject/remapping/dist/types
 import { decode as decode_mappings } from 'sourcemap-codec';
 import { getLocator } from 'locate-character';
 import { StringWithSourcemap, sourcemap_add_offset, combine_sourcemaps } from '../utils/string_with_sourcemap';
+import { compareByGeneratedPositionsInflated } from 'source-map/lib/util';
```

Comment:
I don't think we can do this since this means we now have a dependency on source-map inside the compiler which we want to avoid. If there's no way to patch this we have to live with the slower solution.

Reply:
no problem, [that fn](https://github.com/mozilla/source-map/blob/7a263408de1b60e60b6220ca3e36d9b3e4aa047d/lib/util.js#L126) has only 30 lines of code, which can be reduced to 10 lines

```js
function areEqualMappings(a, b) {
  return (
    a.generatedLine == b.generatedLine &&
    a.generatedColumn == b.generatedColumn &&
    a.source == b.source &&
    a.originalLine == b.originalLine &&
    a.originalColumn == b.originalColumn &&
    a.name == b.name
  );
}
```

Reply:
I actually wonder - why do we want to avoid a dependency on `source-map`? I *think* that Rollup would just bundle that one function that we used and it would not cause any real additional weight, so there should not be any issue with it. I didn't test and could be wrong about that though

Reply:
> a dependency on source-map inside the compiler which we want to avoid.

> why do we want to avoid a dependency on source-map?

poor maintenance / abandonware (mozilla is broke without google), last release (0.7.3) was 3 years ago, [huge bundle size](https://github.com/mozilla/source-map/issues/400) in version 0.8 .... (more reasons?)

areEqualMappings is an optimized version (could be even better), which does matter, cos its called for every mapping (hot code)

Reply:
No the real reason is that it bloats the bundle size as stated in #1863 . Also it contains a WASM part which may or may not be needed and that would also complicate the build setup.

Reply:
there is actually a technical argument for not using the official type:
the import function is using "hidden" properties, that are not defined in [source-map.d.ts](https://github.com/mozilla/source-map/blob/7a263408de1b60e60b6220ca3e36d9b3e4aa047d/source-map.d.ts#L286)

Reply:
I would think Rollup would tree shake everything but the used function. But if we're using an optimized version here then that's reason enough for me to keep it this way.